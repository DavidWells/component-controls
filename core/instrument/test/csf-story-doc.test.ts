import { loadStoriesTests } from './loadTestFiles';

describe('csf-story-doc', () => {
  loadStoriesTests({ stories: { storeSourceFile: false } }, ['csf', 'doc']);
});
