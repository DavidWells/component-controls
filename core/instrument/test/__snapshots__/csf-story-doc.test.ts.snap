// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-story-doc no-title.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "date": 2020-06-17T18:28:16.346Z,
    "dateModified": 2020-06-17T18:28:16.347Z,
    "package": "b9103c46172438afdde76f8345fdfed0",
    "test": 1,
    "title": "Story",
  },
  "packages": Object {
    "b9103c46172438afdde76f8345fdfed0": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.3.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "b9103c46172438afdde76f8345fdfed0",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/doc/no-title.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.3.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Story',
  test: 1,
};
",
}
`;

exports[`csf-story-doc title-and-parameters.js 1`] = `
Object {
  "components": Object {
    "72da0b761debbc67acfdcfe4cc253e29": Object {
      "name": "ControlsTable",
    },
  },
  "doc": Object {
    "component": "ControlsTable",
    "components": Object {
      "ControlsTable": "72da0b761debbc67acfdcfe4cc253e29",
    },
    "date": 2020-06-17T18:28:16.347Z,
    "dateModified": 2020-06-17T18:28:16.347Z,
    "package": "aef0eeece1e59f4690b7e3f0065cdc41",
    "parameters": Object {
      "addonControls": Object {
        "smart": false,
      },
      "component": "ControlsTable",
    },
    "title": "Storybook/Blocks/ControlsTable",
  },
  "packages": Object {
    "aef0eeece1e59f4690b7e3f0065cdc41": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.3.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "aef0eeece1e59f4690b7e3f0065cdc41",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/doc/title-and-parameters.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.3.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Storybook/Blocks/ControlsTable',
  parameters: {
    component: ControlsTable,
    addonControls: {
      smart: false,
    },
  },
};
",
}
`;

exports[`csf-story-doc title-controls-and-parameters.js 1`] = `
Object {
  "components": Object {
    "04b628ec4e86b8383f71524efe46f138": Object {
      "name": "ControlsTable",
    },
  },
  "doc": Object {
    "component": "ControlsTable",
    "components": Object {
      "ControlsTable": "04b628ec4e86b8383f71524efe46f138",
    },
    "date": 2020-06-17T18:28:16.347Z,
    "dateModified": 2020-06-17T18:28:16.347Z,
    "package": "4dd4688f69a323d0b893ca9c902cc47a",
    "parameters": Object {
      "controls": Object {
        "name": Object {
          "label": "Name",
          "order": 9999,
          "type": "ControlTypes.TEXT",
          "value": "Mark",
        },
      },
    },
    "title": "Storybook/Kind",
  },
  "packages": Object {
    "4dd4688f69a323d0b893ca9c902cc47a": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.3.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "4dd4688f69a323d0b893ca9c902cc47a",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/doc/title-controls-and-parameters.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.3.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Storybook/Kind',
  component: ControlsTable,
  parameters: {
    controls: {
      name: {
        type: ControlTypes.TEXT,
        label: 'Name',
        value: 'Mark',
        order: 9999,
      },
    },
  },
};
",
}
`;

exports[`csf-story-doc with-title.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "date": 2020-06-17T18:28:16.347Z,
    "dateModified": 2020-06-17T18:28:16.347Z,
    "package": "44bab54c940e1430b4b259f5bad24b7a",
    "title": "Storybook/Blocks/ControlsTable",
  },
  "packages": Object {
    "44bab54c940e1430b4b259f5bad24b7a": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.3.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "44bab54c940e1430b4b259f5bad24b7a",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/doc/with-title.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.3.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Storybook/Blocks/ControlsTable',
};
",
}
`;
